core/eolearn/tests/test_core_tasks.py

```diff
@@ -71,38 +73,34 @@ def patch_fixture():
     return patch
 
 
-def test_copy(patch):
-    patch_copy = CopyTask().execute(patch)
-
-    assert patch == patch_copy, "Copied patch is different"
-
-    patch_copy.data["new"] = np.arange(1).reshape(1, 1, 1, 1)
-    assert "new" not in patch.data, "Dictionary of features was not copied"
+@pytest.mark.parametrize("task", [DeepCopyTask, CopyTask])
+def test_copy(task: CopyTask, patch: EOPatch) -> None:
+    patch_copy = task().execute(patch)
+    assert deep_eq(patch_copy, patch)
```

Comment:
also why do you compare them with `deep_eq` instead of `==`?

Reply:
`patch.data["bands"]` is np.array usually when compere two np.arrays `numpy.testing.assert_array_equal` is used not `==` applied same logic here might be wrong :thinking:

Reply:
EOPatches are not numpy arrays. You should check how they implement equality